engines/wintermute/utils/path_util.cpp

```diff
@@ -74,6 +74,11 @@ AnsiString PathUtil::getDirectoryName(const AnsiString &path) {
 AnsiString PathUtil::getFileName(const AnsiString &path) {
 	AnsiString newPath = unifySeparators(path);
 	Common::String lastPart = Common::lastPathComponent(newPath, '/');
+
+	if (lastPart.empty()) {
+		return path;
+	}
```

Comment:
> On the other hand, does lastPathComponent actually return strings with the separator still in them? (Or could we simply do else { return lastPart; } ?

I'm sure it should not contain any path separator inside the result. So, returning the result of ``lastPathComponent`` should be fine AFAICT. I do not understand the reason for the `if` check below though.